src/encoded/types/individual.py

```diff
@@ -118,3 +137,114 @@ class IndividualZebrafish(Individual):
     item_type = 'individual_zebrafish'
     schema = load_schema('encoded:schemas/individual_zebrafish.json')
     embedded_list = Individual.embedded_list + ['zebrafish_vendor.name']
+
+
+# validator for individual relations - same species
+def validate_individual_relations(context, request):
+    '''Validates that individual relations are within the same species,
+    limited to two relations per individual (max 1 paternal and 1 maternal),
+    not self relations, and unique (no duplicate relations).
+    '''
+    data = request.json
+    related_individuals = data.get('individual_relation')  # a list of dicts
+    if related_individuals is None:
+        return
+    any_failures = False
+    # check if the individual info is in request (POST) or in context (PATCH)
```

Comment:
In order to save a call to get_item_if_you_can I might change lines 154 to 160 to something like:

    if 'uuid' in data:  # it's a POST
        individual_uuid = data['uuid']
    else:  # get it from the context it's a PATCH
        try:
            individual_uuid = str(context.uuid)
        except Exception:  # would be very wonky if this happened
            pass
        finally:
            data = context.properties  # organism will be here now
    organism_id = data.get('organism')

Reply:
This change is probably optional but another possible better practice is that the try clause in my example includes only a signal statement to try rather than the 2 in your version either of which could potentially cause the exception.  Not sure.